### DISCUSSION TITLE
How to customize the local pickup options rendering

### DISCUSSION THREAD

**Discussion by calind** [woocommerce/woocommerce#45007] [WooCommerce, Blocks, and Site Editing]
Hi,

I'm struggling to add a locker box delivery option to a store. I've managed to add the shipping method as a local delivery option which is fine, but I need to customize the rendered radio option to add a button/link to change the locker box.

<img width="976" alt="Screenshot 2023-07-13 at 19 56 21 copy" src="https://github.com/woocommerce/woocommerce-blocks/assets/266375/c5216a6a-693d-4440-ab61-bf67a5a3d2c0">

Or do you have any suggestion, UI-wise for adding locker box delivery?

Thanks!

**Comment by passatgt** [comment]
I think this needs to be developed as a separate inner block. I'm playing with it, since i have a locker box shipping method that would show a map of many different provider's pickup points. The UI would look like this:

![Screenshot 2023-02-28 at 17 57 26](https://github.com/woocommerce/woocommerce-blocks/assets/43415/912d0ff5-ddf1-402f-b95e-1c0325b68622)

I managed to create an inner block for this and you can add it in the pickup options section by using the "woocommerce/checkout-pickup-options-block" as the parent block. The shop owner will had to set this up manually though, which is not a great UX instead of appearing this automatically, but this is a different topic(missing hooks for example). 

And i'm stuck here, i have no idea how to add the selected point the the current session, how to update the order review section(prices might change based on different providers for example), how to save it to the order etc... If anyone can help me with this, send me a message, i'll pay for a solution.
